compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -262,7 +262,7 @@ pub fn each_linked_rlib(
     };
 
     for &cnum in crates {
-        match fmts.get(cnum.as_usize() - 1) {
+        match fmts.get(cnum) {
```

Comment:
Nothing gets changed here because `crates` only includes dependencies and not the local crate.

Reply:
Suggestion: can we rename this to `dep_crates` or something? Because `crates` to me emcompasses "all the crates", including in particular the local crate, which is not the intention here.